runtime-light/stdlib/array/array-functions.h

```diff
@@ -32,13 +53,51 @@ void f$shuffle(array<T> &arr) noexcept {
 }
 
 template<class T>
-array<array<T>> f$array_chunk(const array<T> &a, int64_t chunk_size, bool preserve_keys = false) {
-  php_critical_error("call to unsupported function");
+task_t<array<T>> f$array_filter(const array<T> &a) noexcept {
+  co_return co_await array_functions_impl_::array_filter_impl(a, [](const auto &it) { return f$boolval(it.get_value()); });
+}
+
+template<class T, class Callback>
+task_t<array<T>> f$array_filter(const array<T> &a, const Callback &callback) noexcept {
+  if constexpr (is_async_function_v<Callback, T>) {
+    co_return co_await array_functions_impl_::array_filter_impl(a, [&callback](const auto &it) -> task_t<bool> {
```

Comment:
What is the idea behind creating a proxy-lambda here?

Reply:
For casting result of callback(T) in bool